From: Seth Vidal <skvidal@fedoraproject.org>
Date: Tue, 25 Jan 2011 17:07:44 +0000 (-0500)
Subject: setup the _repos_persistdir in _getConfig b/c of how
X-Git-Url: http://yum.baseurl.org/gitweb?p=yum.git;a=commitdiff_plain;h=23b9f3855dcc8db217fab9b7ea419f8a400fb696;hp=864b994e538862606183e67e3c794764d18f119a

setup the _repos_persistdir in _getConfig b/c of how
local repos are setup by repoquery and other tools.
fixes rh bug: 672588
---

diff --git a/yum/__init__.py b/yum/__init__.py
index efe6f07..78c71ec 100644
--- a/yum/__init__.py
+++ b/yum/__init__.py
@@ -349,7 +349,10 @@ class YumBase(depsolve.Depsolve):
 
         # who are we:
         self.conf.uid = os.geteuid()
-        
+        # repos are ver/arch specific so add $basearch/$releasever
+        self.conf._repos_persistdir = os.path.normpath('%s/repos/%s/%s/'
+               % (self.conf.persistdir,  self.yumvar.get('basearch', '$basearch'),
+                  self.yumvar.get('releasever', '$releasever')))        
         self.doFileLogSetup(self.conf.uid, self.conf.logfile)
         self.verbose_logger.debug('Config time: %0.3f' % (time.time() - conf_st))
         self.plugins.run('init')
@@ -418,10 +421,7 @@ class YumBase(depsolve.Depsolve):
             else:
                 thisrepo.repo_config_age = repo_age
                 thisrepo.repofile = repofn
-                # repos are ver/arch specific so add $basearch/$releasever
-                self.conf._repos_persistdir = os.path.normpath('%s/repos/%s/%s/'
-                     % (self.conf.persistdir,  self.yumvar.get('basearch', '$basearch'),
-                        self.yumvar.get('releasever', '$releasever')))
+
                 thisrepo.base_persistdir = self.conf._repos_persistdir
 
 
